src/tests/ftest/util/dmg_utils.py

```diff
@@ -23,44 +23,183 @@
+
+        def __init__(self):
+            """Create a dmg Command object."""
+            super(DmgCommand.DmgPrepareSubCommand, self).__init__(
+                "/run/dmg/prepare/*", "prepare")
+            self.pci_wl = FormattedParameter("-w {}")
+            self.hugepages = FormattedParameter("-p {}")
+            self.targetuser = FormattedParameter("-u {}")
+            self.nvmeonly = FormattedParameter("-n", False)
+            self.scmonly = FormattedParameter("-s", False)
+            self.force = FormattedParameter("-f", False)
+            self.reset = FormattedParameter("--reset", False)
+
+
+def storage_scan(hosts):
+    """ Execute scan command through dmg tool to servers provided.
+
+    Args:
+        hosts (list): list of servers to run scan on.
+
+    Returns:
+        Avocado CmdResult object that contains exit status, stdout information.
+
+    """
+    # Create and setup the command
+    dmg = DmgCommand(get_file_path("bin/daos_shell"))
+    dmg.request.value = "storage"
+    dmg.action.value = "scan"
+    dmg.insecure.value = True
+    dmg.hostlist.value = hosts
+
+    try:
+        result = dmg.run()
+    except CommandFailure as details:
+        print("<dmg> command failed: {}".format(details))
+        return False
+
+    return result
+
+
+def storage_format(hosts):
+    """ Execute format command through dmg tool to servers provided.
+
+    Args:
+        hosts (list): list of servers to run format on.
+
+    Returns:
+        Avocado CmdResult object that contains exit status, stdout information.
+
+    """
+    # Create and setup the command
+    dmg = DmgCommand(get_file_path("bin/daos_shell"))
+    dmg.insecure.value = True
+    dmg.hostlist.value = hosts
+    dmg.request.value = "storage"
+    dmg.action.value = "format"
+    dmg._get_action_command()
+    dmg.action_command.force.value = True
+
+    try:
+        result = dmg.run(sudo=True)
+    except CommandFailure as details:
+        print("<dmg> command failed: {}".format(details))
+        return False
+
+    return result
+
+
+def storage_prep(hosts, user=False, hugepages="4096", nvme=False,
+                 scm=False):
+    """Execute prepare command through dmg tool to servers provided.
+
+    Args:
+        hosts (list): list of servers to run prepare on.
+        user (str, optional): User with priviledges. Defaults to False.
+        hugepages (str, optional): Hugepages to allocate. Defaults to "4096".
+        nvme (bool, optional): Perform prep on nvme. Defaults to False.
+        scm (bool, optional): Perform prep on scm. Defaults to False.
+
+    Returns:
+        Avocado CmdResult object that contains exit status, stdout information.
+
+    """
+    # Create and setup the command
+    dmg = DmgCommand(get_file_path("bin/daos_shell"))
+    dmg.insecure.value = True
+    dmg.hostlist.value = hosts
+    dmg.request.value = "storage"
+    dmg.action.value = "prepare"
+    dmg._get_action_command()
```

Comment:
(pylint-protected-access) Access to a protected member _get_action_command of a client class